sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -640,6 +648,19 @@ boolean bucketAccessible(GcsPath path, BackOff backoff, Sleeper sleeper) throws
     }
   }
 
+  /**
+   * Checks whether the GCS bucket exists. Throws exception if the bucket is inaccessible due to
+   * permissions or does not exist.
```

Comment:
Can't it throw for more reasons? https://github.com/apache/beam/blob/6dea0d15d0a97d243a2fe56684c2e193cbea14d2/sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java#L673